keras_retinanet/bin/train.py

```diff
@@ -293,6 +295,7 @@ def csv_list(string):
     parser.add_argument('--tensorboard-dir', help='Log directory for Tensorboard output', default='./logs')
     parser.add_argument('--no-snapshots',    help='Disable saving snapshots.', dest='snapshots', action='store_false')
     parser.add_argument('--no-evaluation',   help='Disable per epoch evaluation.', dest='evaluation', action='store_false')
+    parser.add_argument('--true-shapes',     help='Use actual model to compute shapes of layers used for FPN.', dest='true_shapes', action='store_true', default=False)
```

Comment:
Can you explain why this is necessary? Does VGG not divide the image by a factor 2 every stage? Or does the input of VGG not always allow a "nice" division by 2?

Also, `dest='true_shapes'` is already the default destination. `argparse` replaces all `-` with `_`.

Reply:
VGG uses `padding="valid"` on the pooling layers, that is why the current division by 2 will not work.

Reply:
For just this case of VGG we could also have two different methods of computing the shapes, one for `same` padding and one for `valid`. But then we would also run into trouble for example with AlexNet, which, if I remember correctly, uses a MaxPool of size 3 and stride 2, which will be trickier to cover without using the actual shape from the model.

Reply:
The `true-shapes` script argument itself is only there so that computing shapes based on the model does not implicitly become the new default. Passing the model directly to the Generator is maybe not the best way, we can try to figure out an alternative. In any case, we would still need to pass something into the generator, something that needs the model as an input.